src/main/java/seedu/address/logic/parser/AddAppointmentCommandParser.java

```diff
@@ -36,7 +44,17 @@ public AddAppointmentCommand parse(String args) throws ParseException {
         argMultimap.verifyNoDuplicatePrefixesFor(
                 PREFIX_PATIENT_ID, PREFIX_DATETIME, PREFIX_ATTEND, PREFIX_APPOINTMENT_DESCRIPTION);
         int studentId = ParserUtil.parseIndex(argMultimap.getValue(PREFIX_PATIENT_ID).get()).getOneBased();
+
+        if (!RelationshipUtil.personExists(studentId, patients)) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_PATIENT_ID, studentId)
+             );
+        }
+
         LocalDateTime appointmentDateTime = ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_DATETIME).get());
+        if (appointmentDateTime.isBefore(LocalDateTime.now())) {
```

Comment:
Refering to the conversation on allowing users to add appointments after the date has passed in the feature spec v1.2 document. I thought the conclusion was to allow users to be able to add appointments after the date has passed. Is there a reason why users should not be able to add appointments in after the date has passed?

My suggestion is that we should check if the date and time overlaps with another appointment instead.

Reply:
Agree

Reply:
Thanks for addressing the concern 😃